filebeat/input/filestream/fswatch_test.go

```diff
@@ -251,16 +251,40 @@ scanner:
 		require.Equal(t, loginp.OpDone, e.Op)
 	})
 
+	t.Run("does not emit events for empty files", func(t *testing.T) {
```

Comment:
could we maybe add a few things to the test?

I was thinking it would be nice to make sure that the if condition was hit, so a check to make sure that the warn log file was emitted, and after that make sure that if we write to the file, then events are produced.

Reply:
@leehinman The way we check for "an event is not emitted" is no event after a timeout. And when a timeout hits we have a "marker event" that has `OpDone`.

So, it's very tricky to introduce the test that would check the empty file is picked up after a write later, I'll think about it.

I've tested this scenario manually though, see the description of this PR.

Reply:
I agree the manual test covers this at this point in time.  I'd just like a test for the future.

With OpDone marker event, is there anyway we could get that if the go routine running the fswatcher was starved and didn't execute before the deadline?

FYI using [DevelopmentSetup](https://github.com/elastic/elastic-agent-libs/blob/16bf3dfcb3655978896c7d081b1b4e3dbf967046/logp/core.go#L160) and [ToObserverOutput](https://github.com/elastic/elastic-agent-libs/blob/16bf3dfcb3655978896c7d081b1b4e3dbf967046/logp/options.go#L40) it is pretty easy to get the log files.  So you maybe write 0 length file, wait until the log file shows up, and then write real entry and get the event like "detects a new file" test?

Reply:
@leehinman thanks for the hint, addressed in https://github.com/elastic/beats/pull/36076/commits/513b81326c6fc6b5bdf99be1a536c5fc278ba2fc